modules/utxo-lib/src/bitgo/wallet/WalletOutput.ts

```diff
@@ -29,14 +30,67 @@ export function addWalletOutputToPsbt(
   if (scriptType === 'p2tr' || scriptType === 'p2trMusig2') {
     const payment =
       scriptType === 'p2tr' ? createPaymentP2tr(walletKeys.publicKeys) : createPaymentP2trMusig2(walletKeys.publicKeys);
+    psbt.addOutput({ script: payment.output!, value });
+  } else {
+    const { scriptPubKey: script } = createOutputScript2of3(walletKeys.publicKeys, scriptType);
+    psbt.addOutput({ script, value });
+  }
+  updateWalletOutputForPsbt(psbt, rootWalletKeys, psbt.data.outputs.length - 1, chain, index);
+}
+
+/**
+ * Update the wallet output with the required information when necessary. If the
+ * information is there already, it will skip over it.
+ *
+ * This function assumes that the output script and value have already been set.
+ *
+ * @param psbt the PSBT to update change output at
+ * @param rootWalletKeys keys that will be able to spend the output
+ * @param outputIndex output index where to update the output
+ * @param chain chain code to use for deriving scripts (and to determine script
+ *              type) chain is an API parameter in the BitGo API, and may be
+ *              any valid ChainCode
+ * @param index derivation index for the change address
+ * @param value value of the change output
+ */
+export function updateWalletOutputForPsbt(
+  psbt: UtxoPsbt,
+  rootWalletKeys: RootWalletKeys,
+  outputIndex: number,
+  chain: ChainCode,
+  index: number
+): void {
+  if (psbt.data.outputs.length <= outputIndex) {
+    throw new Error(
+      `outputIndex (${outputIndex}) is too large for the number of outputs (${psbt.data.outputs.length})`
+    );
+  }
+
+  // It is expensive to attempt to compute every output address using psbt.txOutputs[outputIndex]
+  // to then just get the script. Here, we are doing the same thing as what txOutputs() does in
+  // bitcoinjs-lib, but without iterating over each output.
+  const outputScript: Buffer = (psbt as any).__CACHE.__TX.outs[outputIndex].script;
```

Comment:
This is kind of ugly, but it allows us to not call `const outputScript = psbt.txOutputs[outputIndex]` for each iteration. Additionally, we do not even try to make any addresses from the scripts, so it is very efficient

Reply:
let's add a method `getOutputScript(outputIndex): Buffer` to UtxoPsbt then (maybe we can upstream it)